packages/@aws-cdk/aws-certificatemanager/test/certificate.test.ts

```diff
@@ -91,6 +91,37 @@ test('throws when domain name is longer than 64 characters', () => {
   }).toThrow(/Domain name must be 64 characters or less/);
 });
 
+test('does not throw when domain name is longer than 64 characters with tokens', () => {
+  const stack = new Stack();
+  const baseDomain = `${'123456789.'.repeat(5)}example.com`; //61 characters
+  //84 ${Token[AWS.Region.7]}.123456789.123456789.123456789.123456789.123456789.example.com"
+  const domainName = `${Aws.REGION}.${baseDomain}`;
+  new Certificate(stack, 'Certificate', {
+    domainName,
+    validation: CertificateValidation.fromEmail({
+      [domainName]: 'example.com',
+    }),
+  });
```

Comment:
```suggestion
  const embededToken = '${Token[AWS.Region.7]}';
  const domainName = `${embededToken}${'a'.repeat(65-embededToken.length)}`;
  new Certificate(stack, 'Certificate', {
    domainName,
    validation: CertificateValidation.fromEmail({
      [domainName]: 'example.com',
    }),
  });
```

Reply:
The embededToken must be real reference, otherwise synthesis will fail, since it cannot find unstringified token, registered when token was stringified. So using stringified version sadly did not work.

Did apply the simplified repeat in this test. Which is indeed much more readable.
And also a sligthly modified version in the dns-validated-certificate.test.ts since it checks the provided zone is authoritive for the provided domain.